src/main/java/com/andreamazzarella/contact_manager/Age.java

```diff
@@ -0,0 +1,36 @@
+package com.andreamazzarella.contact_manager;
+
+import java.util.regex.Pattern;
+
+public class Age {
+    private final String age;
+
+    public Age(String age) {
+        age = removeAllWhiteSpace(age);
+
+        if (isValidNumber(age)) {
```

Comment:
Thank you, that makes sense, and I think it will also act as one layer of validation on the front end. At a first layer things may fail if you enter invalid data (such as a negative number for age), and then, once you have a legal contact, you get contact validation made by the AddContact transaction, for things like minimum age.

I wonder if however this creates a nasty pattern of try/catches on the front end like this:

```
private Contact createContactFromFields() {
        try {
            Age age = new Age(newAge.getText());
        } catch { COMMUNICATE WHY IT FAILED }

// potentially add one try catch per field...so disgusting! And maybe they need to be nested too?

        String firstName = newFirstName.getText();
        String lastName = newLastName.getText();
        String streetAddress = newStreetAddress.getText();
        String postalCode = newPostalCode.getText();
        TelephoneNumber telephoneNumber = new TelephoneNumber(newTelephoneNumber.getText());
      

        return new Contact(firstName, lastName, streetAddress, postalCode, telephoneNumber, age);
    }
```